src/BenchmarkDotNet/Toolchains/MonoWasm/WasmExecutor.cs

```diff
@@ -0,0 +1,146 @@
+                CustomDotNetCliPath,
+                artifactsPaths.BinariesDirectoryPath,
+                artifactsPaths.ProgramName,
+                benchmarkId.ToArguments(),
+                redirectStandardInput: true);
+
+
+            startInfo.SetEnvironmentVariables(benchmarkCase, resolver);
+
+            using (var process = new Process { StartInfo = startInfo })
+            using (new ConsoleExitHandler(process, logger))
+            {
+                var loggerWithDiagnoser = new SynchronousProcessOutputLoggerWithDiagnoser(logger, process, diagnoser, benchmarkCase, benchmarkId);
+
+                logger.WriteLineInfo($"// Execute: {process.StartInfo.FileName} {process.StartInfo.Arguments} in {process.StartInfo.WorkingDirectory}");
+
+                diagnoser?.Handle(HostSignal.BeforeProcessStart, new DiagnoserActionParameters(process, benchmarkCase, benchmarkId));
+
+                process.Start();
+
+                process.EnsureHighPriority(logger);
+                if (benchmarkCase.Job.Environment.HasValue(EnvironmentMode.AffinityCharacteristic))
+                {
+                    process.TrySetAffinity(benchmarkCase.Job.Environment.Affinity, logger);
+                }
+
+                loggerWithDiagnoser.ProcessInput();
+                string standardError = process.StandardError.ReadToEnd();
+
+                process.WaitForExit(); // should we add timeout here?
+
+                if (process.ExitCode == 0)
+                {
+                    return new ExecuteResult(true, process.ExitCode, process.Id, loggerWithDiagnoser.LinesWithResults, loggerWithDiagnoser.LinesWithExtraOutput);
+                }
+
+                if (!string.IsNullOrEmpty(standardError))
+                {
+                    logger.WriteError(standardError);
+                }
+
+                return new ExecuteResult(true, process.ExitCode, process.Id, Array.Empty<string>(), Array.Empty<string>());
+            }
+        }
+
+
+        internal static ProcessStartInfo BuildStartInfo(string customDotNetCliPath,
+                                                        string workingDirectory,
+                                                        string programName,
+                                                        string benchmarkId,
+                                                        IReadOnlyList<EnvironmentVariable> environmentVariables = null,
+                                                        bool redirectStandardInput = false)
+        {
+            const string dotnetMultiLevelLookupEnvVarName = "DOTNET_MULTILEVEL_LOOKUP";
+
+            ProcessStartInfo startInfo = new ProcessStartInfo
+            {
+                FileName = "v8",
+                WorkingDirectory = workingDirectory,
+                Arguments = $"--expose_wasm runtime.js  -- --enable-gc --run {programName}.dll {benchmarkId} ",
+                UseShellExecute = false,
```

Comment:
it looks that we could just extend the `Executor.CreateStartInfo` with handling of the `Wasm` runtime and avoid having a new dedicated `WasmExecutor` type

https://github.com/dotnet/BenchmarkDotNet/blob/0b0ede9565758d16f774c68bb02b8fbd7feea576/src/BenchmarkDotNet/Toolchains/Executor.cs#L104-L115

Reply:
I don't think we can do this now that I added the javaScriptEngineArguments option, since some arguments have to go to the engine, and others to the runtime. Or, at least, it doesn't look like the existing pattern?

Reply:
to achieve that you could move all the fields from `WasmSettings` type to `WasmRuntime` type and create such runtime when parsing the args in ConfigParser:

https://github.com/dotnet/BenchmarkDotNet/blob/50e5157d0fc5c67e8982c9ff4817c511684dae49/src/BenchmarkDotNet/ConsoleArguments/ConfigParser.cs#L362-L368

we can leave it as it is and refactor in a separate PR